crates/cext/src/transpiler/transpile_layout.rs

```diff
@@ -0,0 +1,507 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::const_ptr_as_ref;
+use qiskit_transpiler::transpile_layout::TranspileLayout;
+
+/// @ingroup QkTranspileLayout
+/// Return the number of qubits in the input circuit to the transpiler.
+///
+/// @param layout A pointer to the ``QkTranspileLayout``.
+///
+/// @return The number of input qubits
+///
+/// # Safety
+///
+/// Behavior is undefined if ``layout`` is not a valid, non-null pointer to a
+/// ``QkTranspileLayout``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_transpile_layout_num_input_qubits(
+    layout: *const TranspileLayout,
+) -> u32 {
+    let layout = unsafe { const_ptr_as_ref(layout) };
+    layout.num_input_qubits()
+}
+
+/// @ingroup QkTranspileLayout
+/// Return the number of qubits in the output circuit from the transpiler.
+///
+/// @param layout A pointer to the ``QkTranspileLayout``.
+///
+/// @return The number of output qubits
+///
+/// # Safety
+///
+/// Behavior is undefined if ``layout`` is not a valid, non-null pointer to a
+/// ``QkTranspileLayout``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_transpile_layout_num_output_qubits(
+    layout: *const TranspileLayout,
+) -> u32 {
+    let layout = unsafe { const_ptr_as_ref(layout) };
+    layout.num_output_qubits()
+}
+
+/// @ingroup QkTranspileLayout
+/// Return the initial layout of a ``QkTranspileLayout``.
```

Comment:
Would "query" be more accurate since it doesn't "return" the layout but instead writes it into the variable passed by the user? (Same for the other functions below)
```suggestion
/// Query the initial layout of a ``QkTranspileLayout``.
```